assets/src/js/listing_pages.js

```diff
@@ -17,4 +20,36 @@ export const setupListingPages = () => {
 
   listViewToggle.onclick = switchViews;
   gridViewToggle.onclick = switchViews;
+
+  // Setup filters for the News & Stories page.
+  const filters = document.querySelector('.listing-page-filters');
+  if (!filters) {
+    return;
+  }
+
+  // Functions and constants.
+  const CATEGORY_FILTER = 'category';
+
+  const updateFilter = event => {
+    const {target: {id, value}} = event;
+    const newUrl = new URL(window.location.href.split('/page/')[0]);
+
+    if (value) {
+      newUrl.searchParams.set(id, value);
+    } else {
+      newUrl.searchParams.delete(id);
+    }
+    window.location.href = newUrl.href;
+  };
+
+  // Category filter.
+  document.getElementById(CATEGORY_FILTER).onchange = updateFilter;
+
+  // Add 'No posts found' text when needed.
+  if (!listingPageContent.querySelector('.wp-block-post-template')) {
```

Comment:
I'm sorry @mleray, but I am not a big fan of this solution.
Cannot we handle this on the server side?
Maybe we can modify the `get_categories` function used in the `listing-page-filters.twig` file to get only categories from posts, not pages. What do you think?

Reply:
You're right, updated!

Reply:
Now, if I try to access a category that exists but has no associated posts, I see an empty list.
Check it: https://www-dev.greenpeace.org/test-oberon/news-stories/?category=greenpeace

Reply:
But the rest of the categories behave differently:
https://www-dev.greenpeace.org/test-oberon/news-stories/?category=fake-category

Reply:
Hm 🤔 What do you think we should do in this case? Add the "No posts found" message back?

Reply:
For consistency, I'd keep the same behavior as a nonexistent category: show all the posts and select the "All topics" option in the dropdown.

Reply:
That might be tricky, but I'll look into it

Reply:
Updated!